src/ApimEventProcessor/MoesifHttpMessageProcessor.cs

```diff
@@ -84,7 +121,30 @@ public async Task<List<EventModel>> BuildMoesifEvents(Dictionary<Guid, KeyValueP
             List<EventModel> events = new List<EventModel>();
             foreach(KeyValuePair<HttpMessage, HttpMessage> kv in completedMessages.Values)
             {
-                events.Add(await BuildMoesifEvent(kv.Key, kv.Value));
+                var moesifEvent = await BuildMoesifEvent(kv.Key, kv.Value);
+                try {
+                    // Get Sampling percentage
+                    samplingPercentage = appConfig.getSamplingPercentage(config, moesifEvent.UserId, moesifEvent.CompanyId);
+                    Random random = new Random();
+                    double randomPercentage = random.NextDouble() * 100;
+                    if (samplingPercentage >= randomPercentage)
+                    {
+                        moesifEvent.Weight = appConfig.calculateWeight(samplingPercentage);
+                        // Add event to the batch
+                        events.Add(moesifEvent);
+                        if (lastWorkerRun.AddMinutes(5) < DateTime.UtcNow)
```

Comment:
hmm, I thought we discussed to check for configEtag after posting events to moesif, if different then fetch the config. Any reason you are doing every 5 mins? In kong we had all sort of problems, so we did periodically there.

Reply:
If we check after every event, it'd be minute delay since we fetch/compare the eTag and if needed we'll have to fetch it. I thought about doing that but realize it'd be better to do it every 5 min incase we couldn't fetch the config or tag for some reason, we don't want to keep making a call after every batch is sent.

Reply:
Currently, don't we wait for the response after sending batch of events asynchronously?

Reply:
looks like we dont look for response now. Its fine then leave it as it is